_docs/LitmusVagrantTesting.md

```diff
@@ -0,0 +1,49 @@
+---
+layout: page
+title: Testing Puppet Modules with Litmus and Vagrant
+---
+
+## Testing Puppet Modules with Litmus and Vagrant
+ 
+### The Setup 
+
+For this demo I'm gonna run through the entire process from start to finish, I hope that you find it useful.
+To start with the first thing that you need to do is clone down the module that you want to test, in this case we'll be using `puppetlabs-scheduled_task`. This can easily be accomplished with a simple command:
+
+```
+git clone https://github.com/user-name/puppetlabs-scheduled_task.git
+```
+
+Now that you have the module cloned down simply run the below command to set up the environment:
+
+```
+bundle install --path .bundle/gems/
+```
+
+### Provisioning the Machine
+
+Now that the environment is setup it is time to provision the machine, while normally you would do this with a vagrant command, Litmus can do it for you if provided with the correct information, as shown below:
+
+![Litmus Vagrant Testing - provisioning the machine](Vagrant%20Provision.png)
+
+If you are testing against multiple different versions then you can also utilise the provision_list command, which let's you set out an entire suite of OS to test against by adding them to and then calling them from the `provision.yml`.
+
+With the machine provisioned it is a simple matter of running the `install_agent` and `install_module` commands and you will be setup and ready to go. With `install_agent` you can run the command as it is to install a default version of the agent or you can specify puppet 5 or 6 specificilly.
+
+![Litmus Vagrant Testing - installing the agent and the module](Agent%20Module%20Install.png)
+
+### Running the Tests
+
+Now that the machine is fully setup it is time to run the tests.
+
+This can be done in several way's, either through Litmus or through rspec, we'll start with the command `bundle exec rake litmus:acceptance:parallel`. 
+
+When this command is given Litmus will retrieve each machine that has been setup and run the test suite against them simultaniusly. This is an excellent function for testing prior to a release or merge as it helps achieve full coverage in a fraction of the time that it would normally take.
+
+If however you are still in the development phase of your work or working on a fix for a specific OS then you may wish to test you changes against a single box. In which case you can do sby providing the name of the machine that you want to run your test's against, as shown here `rake litmus:acceptance:127.0.0.1:55985`.
+
+Finally if you wish to debug your test's with a little more detail or run a specific test file instead of the fully suite, you can use the older, but still functional, rspec command shown here `TARGET_HOST=127.0.0.1:55985 bundle exec rspec spec/acceptance/should_create_task_spec.rb`.
```

Comment:
```suggestion
Finally if you wish to debug your tests with a little more detail or run a specific test file instead of the full suite, you can use the older, but still functional, rspec command shown here `TARGET_HOST=127.0.0.1:55985 bundle exec rspec spec/acceptance/should_create_task_spec.rb`.
```

Reply:
Changed to:
The final command that we wll be covering is `TARGET_HOST=127.0.0.1:55985 bundle exec rspec spec/acceptance/should_create_task_spec.rb`.

Rather than calling on Litmus this instead utilise the older, but still functional, rspec command. This can allow you to specify specific test files to be run rather than the entire suite as well as allowing more detailed debugs to be returned.